lib/core/src/system/web/dom/html/html_scene.rs

```diff
@@ -0,0 +1,76 @@
+//! This module contains `HTMLScene`, a struct to hold `HTMLObject`s.
+
+use crate::prelude::*;
+
+use super::HTMLObject;
+use crate::display::object::DisplayObjectData;
+use data::opt_vec::*;
+
+
+
+// =============
+// === Scene ===
+// =============
+
+/// A collection for holding 3D `Object`s.
+pub struct HTMLScene {
+    display_object : DisplayObjectData,
+    objects        : OptVec<HTMLObject>
+}
+
+impl HTMLScene {
+    /// Searches for a HtmlElement identified by id and appends to it.
+    pub fn new(logger:Logger) -> Self {
+        let display_object = DisplayObjectData::new(logger);
+        let objects        = default();
+        Self{display_object,objects}
+    }
+
+    /// Moves a HTMLObject to the Scene and returns an index to it.
+    pub fn add(&mut self, object: HTMLObject) -> Ix {
+        self.display_object.add_child(&object.display_object);
+        self.objects.insert(object)
+    }
+
+    /// Removes and retrieves a HTMLObject based on the index provided by
+    pub fn remove(&mut self, index: Ix) -> Option<HTMLObject> {
+        let object = self.objects.remove(index);
+        if let Some(object) = &object {
+            self.display_object.remove_child(&object.display_object);
```

Comment:
If `HtmlScene` is display object, then you should do `self.remove_child`. There should not be need to redirect it to a field. Display objects add thsi method automatically to things which implement some trait - see display object definition for more info,

Reply:
As decided on Discord, this will be part of the World integration task.